Server/CommunicationHandler.cs

```diff
@@ -169,17 +179,19 @@ private static void HandleRequestGroups(Connection connection)
         /// <param name="message">The message containing the client Id and group Id.</param>
         private static void HandleJoinGroup(Connection connection, Message message)
         {
-            // TODO: replace by database calls.
+            // TODO: replace by database calls and unsuccesful response.
 
             // Get client by client Id.
             Client clientToAdd = Server.Instance.GetClient(message.ClientId);
 
             // Get the group and add the client to it.
-            Server.Instance.GetGroup(message.GroupID).AddClient(clientToAdd);
+            Group groupToJoin = Server.Instance.GetGroup(message.GroupID);
+            groupToJoin.AddClient(clientToAdd);
 
```

Comment:
Before adding the clientToAdd to the groupToJoin, first check if both are not null?